diff --git a/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp b/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp
index 218c73add..7a40e0e83 100644
--- a/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp
+++ b/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp
@@ -42,7 +42,7 @@ namespace Ogre {
         String GLSLESProgramWriter::TargetLanguage =  "glsles";
 
         // Uniform comparer
-        struct CompareUniformByNameES : std::binary_function<UniformParameterPtr, String, bool>
+        struct CompareUniformByNameES
         {
             bool operator()( const UniformParameterPtr& uniform, const String& name ) const 
             {
@@ -436,7 +436,7 @@ namespace Ogre {
                             // If its not a varying param check if a uniform is written
                             if(!isVarying)
                             {
-                                UniformParameterList::const_iterator itFound = std::find_if( parameterList.begin(), parameterList.end(), std::bind2nd( CompareUniformByNameES(), paramName ) );
+                                UniformParameterList::const_iterator itFound = std::find_if( parameterList.begin(), parameterList.end(), [&paramName](const UniformParameterPtr& uniform) { return uniform->getName() == paramName; } );
                                 if(itFound != parameterList.end())
                                 {	
                                     // Declare the copy variable
diff --git a/Components/RTShaderSystem/src/OgreShaderGLSLProgramWriter.cpp b/Components/RTShaderSystem/src/OgreShaderGLSLProgramWriter.cpp
index 619e6bd2d..1c16a1e93 100644
--- a/Components/RTShaderSystem/src/OgreShaderGLSLProgramWriter.cpp
+++ b/Components/RTShaderSystem/src/OgreShaderGLSLProgramWriter.cpp
@@ -36,7 +36,7 @@ namespace RTShader {
 String GLSLProgramWriter::TargetLanguage = "glsl";
 
 // Uniform comparer
-struct CompareUniformByName : std::binary_function<UniformParameterPtr, String, bool>
+struct CompareUniformByName
 {
 	bool operator()( const UniformParameterPtr& uniform, const String& name ) const 
 	{
@@ -254,7 +254,7 @@ void GLSLProgramWriter::writeSourceCode(std::ostream& os, Program* program)
 					// If its not a varying param check if a uniform is written
 					if(!isVarying)
 					{
-						UniformParameterList::const_iterator itFound = std::find_if( parameterList.begin(), parameterList.end(), std::bind2nd( CompareUniformByName(), paramName ) );
+						UniformParameterList::const_iterator itFound = std::find_if( parameterList.begin(), parameterList.end(), [&paramName](const UniformParameterPtr& uniform) { return uniform->getName() == paramName; } );
 						if(itFound != parameterList.end())
 						{	
 							// Declare the copy variable
diff --git a/OgreMain/src/OgreLodStrategy.cpp b/OgreMain/src/OgreLodStrategy.cpp
index df12e8ac3..a10ba42ae 100644
--- a/OgreMain/src/OgreLodStrategy.cpp
+++ b/OgreMain/src/OgreLodStrategy.cpp
@@ -88,8 +88,7 @@ namespace Ogre {
         return true;
     }
     //---------------------------------------------------------------------
-    struct LodUsageSortLess :
-    public std::binary_function<const MeshLodUsage&, const MeshLodUsage&, bool>
+    struct LodUsageSortLess
     {
         bool operator() (const MeshLodUsage& mesh1, const MeshLodUsage& mesh2)
         {
@@ -103,8 +102,7 @@ namespace Ogre {
         std::sort(meshLodUsageList.begin(), meshLodUsageList.end(), LodUsageSortLess());
     }
     //---------------------------------------------------------------------
-    struct LodUsageSortGreater :
-    public std::binary_function<const MeshLodUsage&, const MeshLodUsage&, bool>
+    struct LodUsageSortGreater
     {
         bool operator() (const MeshLodUsage& mesh1, const MeshLodUsage& mesh2)
         {
diff --git a/OgreMain/src/OgreZip.cpp b/OgreMain/src/OgreZip.cpp
index 7bfe8eee4..8191e113a 100644
--- a/OgreMain/src/OgreZip.cpp
+++ b/OgreMain/src/OgreZip.cpp
@@ -260,7 +260,7 @@ namespace Ogre {
         return ret;
     }
     //-----------------------------------------------------------------------
-	struct FileNameCompare : public std::binary_function<FileInfo, String, bool>
+	struct FileNameCompare
 	{
 		bool operator()(const Ogre::FileInfo& lhs, const String& filename) const
 		{
@@ -278,7 +278,7 @@ namespace Ogre {
 			cleanName = tokens[tokens.size() - 1];
 		}
 
-		return std::find_if (mFileList.begin(), mFileList.end(), std::bind2nd<FileNameCompare>(FileNameCompare(), cleanName)) != mFileList.end();
+		return std::find_if (mFileList.begin(), mFileList.end(), [&cleanName](const FileInfo& info) { return info.filename == cleanName; }) != mFileList.end();
 	}
 	//---------------------------------------------------------------------
 	time_t ZipArchive::getModifiedTime(const String& filename)
